ios/Classes/AblyFlutterPlugin.m

```diff
@@ -39,6 +39,44 @@ -(nullable AblyFlutter *)ablyWithHandle:(NSNumber *)handle;
     [plugin registerWithCompletionHandler:result];
 };
 
+static FlutterHandler _createRestWithOptions = ^void(AblyFlutterPlugin *const plugin, FlutterMethodCall *const call, const FlutterResult result) {
+    AblyFlutterMessage *const message = call.arguments;
+    LOG(@"message for handle %@", message.handle);
+    AblyFlutter *const ably = [plugin ablyWithHandle:message.handle];
+    // TODO if ably is nil here then an error response, perhaps? or allow Dart side to understand null response?
+    result([ably createRestWithOptions:message.message]);
+};
+
+static FlutterHandler _publishRestMessage = ^void(AblyFlutterPlugin *const plugin, FlutterMethodCall *const call, const FlutterResult result) {
+    AblyFlutterMessage *const message = call.arguments;
+    LOG(@"message for handle %@", message.handle);
+    AblyFlutter *const ably = [plugin ablyWithHandle:message.handle];
+    AblyFlutterMessage *const messageData = message.message;
+    NSMutableDictionary<NSString *, NSObject *>* _dataMap = messageData.message;
+    NSString *channelName = (NSString*)[_dataMap objectForKey:@"channel"];
+    NSString *eventName = (NSString*)[_dataMap objectForKey:@"name"];
+    NSObject *eventData = (NSString*)[_dataMap objectForKey:@"message"];
+    ARTRest *client = [ably getRest:messageData.handle];
+    ARTRestChannel *channel = [client.channels get:channelName];
+
+    //Handling nil cases, to be generalized based on other encounters
+    NSString *nilStr = nil;
+    if([eventName isKindOfClass:[NSNull class]]){ eventName = nilStr; }
+    if([eventData isKindOfClass:[NSNull class]]){ eventData = nilStr; }
```

Comment:
@QuintinWillison any scope for improvement here?

As `nil` is not a valid argument for `name` or `data`, we need to pass on an argument of type `NSString`, `NSData`, `NSArray` or `NSDictionary`

Reply:
`objectForKey:` will return `nil` if there is no entry in the data map. Surely you could just not pack an entry on the Dart side?

Also... potentially about to disappear given my suggestion but no need for the `nilStr` intermediary here. `nil` is designed to be assigned directly to any type of Objective-C instance pointer (a.k.a. `id`).

Reply:
@QuintinWillison we do send an entry from dart side, but that entry would be null as passed by package users.

Ex: `restInstance.publish('name', 'message')` will pass both values, but `restInstance.publish()` will pass an map with `{"message": null, "data": null}` which will be available as `nil` in iOS side.

`nil` is not accepted by publish method, as the encoding fails
argument passed: `must be NSString, NSData, NSArray or NSDictionary`

Here is link to the log which states the error:
https://ably-real-time.slack.com/files/U0118SXS2VD/F012UMW8DCZ/untitled

Reply:
That makes sense. However I was imagining that you had control, in Dart land, of the production of the data that ends up causing `_dataMap` to be populated once in Objective-C land. Therefore if the Dart code knows that it needs to set a value of `null` for the `name` key then it can simply not pack a value for that key at all. Or am I missing something?

Reply:
How would it help if we don't send the key to Obj-C side? `publish` method still expects values and publishing fails at encoding stage.

Reply:
@QuintinWillison as discussed, this is sorted in https://github.com/ably/ably-flutter/commit/6ce0328481388fb75de3f95b21b6306b32f8c723 bu not packing the key form dart side